components/collection/index.js

```diff
@@ -0,0 +1,247 @@
+          <Pager
+            currentPage={pagerStatus.pageIndex}
+            totalPages={pagerStatus.totalPages}
+            onClick={pagerAction.setPageIndex}
+            className="mb-2 flex justify-center sm:mb-0 sm:block sm:justify-start"
+          />
+        )}
+      </div>
+    </div>
+  );
+};
+
+CollectionHeader.propTypes = {
+  // Collection pager status
+  pagerStatus: PropTypes.exact({
+    // Number of items per page
+    itemsPerPage: PropTypes.number.isRequired,
+    // Index of the page of items to display
+    pageIndex: PropTypes.number.isRequired,
+    // Total number of pages
+    totalPages: PropTypes.number.isRequired,
+  }).isRequired,
+  // Actions for the current page number and number of items per page
+  pagerAction: PropTypes.exact({
+    // Set the number of items the user wants to see per page
+    setItemsPerPage: PropTypes.func.isRequired,
+    // Set the current page index
+    setPageIndex: PropTypes.func.isRequired,
+  }).isRequired,
+  addSpec: PropTypes.exact({
+    // Object type to append to the Add button label
+    label: PropTypes.string.isRequired,
+    // Path to add the object
+    path: PropTypes.string.isRequired,
+  }),
+};
+
+/**
+ * Display either a list or report view of the collection. For a list, the `children` provides the
+ * content. For the table view, the content comes from this use of the <CollectionTable> component,
+ * and `children` isn’t used. `pagerStatus` holds the current page number and number of items per
+ * page. The list view takes care of these and so `pagerStatus` only gets used for the table view,
+ * which uses `pagerStatus` to display only one page of items while sorting the entire collection.
+ */
+const CollectionContent = ({ collection, pagerStatus, children }) => {
+  // Collection view setting and /profiles content
+  const { collectionView } = useContext(GlobalContext);
+  // True if the user has selected the list view
+  const isListView =
+    collectionView.currentCollectionView === COLLECTION_VIEW_LIST;
+  // True if the user has selected the table view
+  const isTableView =
+    collectionView.currentCollectionView === COLLECTION_VIEW_TABLE;
+
+  useEffect(() => {
+    // If the page loads with a URL that specifies hidden columns, set the table view.
+    const hashedHiddenColumns = extractHiddenColumnIdsFromUrl(
+      window.location.href
+    );
+    if (hashedHiddenColumns?.length >= 0) {
+      collectionView.setCurrentCollectionView(COLLECTION_VIEW_TABLE);
+    }
+  }, [collectionView]);
+
+  if (isListView) {
+    // Display list view.
+    return (
+      <>
+        <CollectionCount count={collection.length} />
+        <div>{children}</div>
+      </>
+    );
+  }
+
+  if (isTableView) {
+    return (
+      <CollectionTable collection={collection} pagerStatus={pagerStatus} />
+    );
+  }
+
+  // No profiles loaded or no collection type in the collection data.
+  return <NoContent>No displayable collection data</NoContent>;
```

Comment:
We might have other collection views in the future, as we do with ENCODE with the matrix view. I feel like it works the way you describe now — if we have content, it goes to the currently selected view, but I might misunderstand.

Reply:
Oh I was just saying that since `COLLECTION_VIEW_LIST` and `COLLECTION_VIEW_TABLE` are strings it's (within the types) possible for `isListView` and `isTableView` to be false but we still have content. And the final return suggests that we should have no content if we get to `<NoContent>...`. And it makes sense that the `CollectionView` will guarantee that it's only one or the other of those two strings, this part of the code just doesn't "know" that. So that we only have the `<NoContent>` case happen when there really is no content I was suggesting we could detect that directly, and then in the case we have content we go through the different views.

Reply:
Sorry, I’m still not sure what you mean. But since I didn’t touch this code for this ticket, you could write a new ticket describing what change you’d like to see.